### PR TITLE
charts/portal - Option to force a redeploy

### PR DESCRIPTION
**Pull request by scottchc1**
**Description of the change**
Conditionally add a timestamp annotation to each deployment template when the option "forceRedeploy" is true. The added/changed annotation guarantee kubernetes to see a difference and redeploy the pod.

forceRedeploy option applies to deployments and statefulsets of the main portal and druid components. it does NOT apply to:
- thirdparty subchart components, e.g. rabbitmq, ingress-nginx, mysql (cannot edit chart definition)
- jobs, e.g. db-upgrade, rbac-upgrade, tls-manager (jobs are immutable anyway)

**Benefits**
This is useful for changes that isn't identified by kubeternes as a difference to redeploy pod when running `helm upgrade`, e.g. updated image on same tag @latest, or a single environment value change in configmap, etc.

**Drawbacks**
- Additional user inputs. Blindly setting forceRedeploy to true for all deployment components will result in max rolling update time.
- The first time changing forceRedeploy from true to false will still incur a redeploy (because changing the annotation to empty from some value is still a change). maintaining false in subsequent upgrade will be fine. we'll have to live with this limitation.

**Applicable issues**
n/a

**Additional information**
sample values edit
```
analytics:
  forceRedeploy: true
apim:
  forceRedeploy: true
authenticator:
  forceRedeploy: true
dispatcher:
  forceRedeploy: true
portalData:
  forceRedeploy: true
portalEnterprise:
  forceRedeploy: true
pssg:
  forceRedeploy: true
solr:
  forceRedeploy: true
tenantProvisioner:
  forceRedeploy: true
druid:
  forceRedeploy: true
```

**Checklist** 
- [] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[charts/gateway]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

**Comment by scottchc1** [comment]
draft for review; not merge ready, not fully updated - readme, values-production

File: charts/portal/values-sample-redeploy.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+global:
```
Comment (hydak) [inline charts/portal/values-sample-redeploy.yaml]: Why did we create a new file here?  It has internal endpoints as well, which we don't want exposed in our codebase

**Review by hydak — COMMENTED** [review]
You need to include the changes to the values-production.yaml .  Also, why isn't rabbitmq included in this forceRedeploy?

File: charts/portal/values-sample-redeploy.yaml (hunk 1 above)
Comment (scottchc1) [reply charts/portal/values-sample-redeploy.yaml]: i'm just showing example of the values to overwrite. this file will be removed before merge

**Comment by scottchc1** [comment]
ready for approval & merge. please review

**Comment by scottchc1** [comment]
NOTE: merge only after 5.0.2

**Pull request merged**